subprojects/docs/src/docs/userguide/upgrade_version_4.adoc

```diff
@@ -0,0 +1,493 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[upgrade_version_4]]
+= Upgrading Gradle Version 4.x to 5.0
+
+This chapter provides the information you need to migrate your older Gradle 4.x builds to the latest version. In most cases, you will need to apply the changes from all versions that come after the one you're upgrading from. For example, if you're upgrading from Gradle 4.3, you will also need to apply the changes from 4.4, 4.5, etc.
+
+A few pointers about upgrading:
+
+ * Run `gradle help` with the version of Gradle you're upgrading to in order to see any deprecations that apply to your build.
+ * Make sure you update the versions of plugins you use in your build when upgrading the Gradle version. Some plugins will break with newer versions of Gradle.
+ * In most cases, deprecated features, properties and methods will be removed in Gradle 5.0, resulting in build errors if you try to use them.
+
+The most significant changes include the following:
```

Comment:
I like this concept of "significant changes" at the top. I wonder if this summary belongs in a more prominent place and make it very visual, such as on gradle.org. 

* The improved POM support will (probably) be enabled by default in Gradle 5.0, thus users may want to upgrade to Gradle 4.10, enable it, then take away that "feature preview" line after upgrading to 5.0.
* I wonder if it would be worthwhile to group "stuff that might break" a little separately from "stuff you should adopt".

Reply:
I can see value in putting some of the significant changes in marketing material, including on gradle.org. I think it should still remain in some form in this chapter.

I don't have an opinion on the approach users should take with the improved POM support. I guess it allows them to isolate one potential source of breaking changes, but does that gain them much over just upgrading to 5.0 and dealing with the issues one by one?

I agree that keeping new features and breaking changes distinct would be helpful. I'll look at that.